File: Settembre/36.ThermostatAdjuster-15Settembre.js
//todo Data la temperatura attuale di una stanza e una temperatura target, restituisci una stringa che indica come regolare la temperatura della stanza in base a questi vincoli:

//! Restituisci "heat" se la temperatura attuale è inferiore al target.
//! Restituisci "cool" se la temperatura attuale è superiore al target.
//! Restituisci "hold" se la temperatura attuale è uguale al target.

//? Tempo di esecuzione:
//* 1 Minuo e 14 Secondi

//?Codice scritto senza utilizzo di ChatGpt

function adjustThermostat(temp, target) {

  if(temp < target) return "heat"
  if(temp > target) return "cool"
  if(temp === target) return "hold"

  return temp;
}

console.log(adjustThermostat(68, 72))

//?Codice scritto da ChatGpt
//* Non ho usato Chat

//* ✅ PRO

//* Logica corretta e semplice

//* Gestisce tutti e tre i casi (<, >, ===)

//* Funzione leggibile e chiara

//* Testa correttamente l’esempio

//! ❌ CONTRO

//! L’ultimo return temp; non serve mai perché tutti i casi sono già coperti

//! Puoi anche usare else if per rendere leggermente più chiaro

//? Versione Chat Gpt 

function adjustThermostat(temp, target) {
  if (temp < target) return "heat";
  else if (temp > target) return "cool";
  else return "hold";
}

console.log(adjustThermostat(68, 72)); // "heat"

//todo Considerazioni: 

//! N.B.

File: Python/2.Settembre/36.Thermostat_adjuster-15Settembre.py
# =========================================================
# FILE TEMPLATE - ANALISI ESERCIZI PYTHON
# Incolla direttamente questo file nel progetto VS Code
# =========================================================

# //Prompt ChatGpt: questo è il modello. Invierò consegna esercizio e codice, e vorrei feedback strutturato

# =========================================================
# //todo Consegna dell'esercizio
# =========================================================
# Data la temperatura attuale di una stanza e una temperatura target, restituisci una stringa che indichi come regolare la temperatura della stanza in base ai seguenti vincoli:

# Restituisci "riscalda" se la temperatura attuale è inferiore alla temperatura target.

# Restituisci "raffredda" se la temperatura attuale è superiore alla temperatura target.

# Restituisci "mantieni" se la temperatura attuale è uguale alla temperatura target.

# =========================================================
# //* Codice scritto senza utilizzo di ChatGpt
# =========================================================

def adjust_thermostat(temp, target):

    match temp:
        case temp if temp < target:
            return "heat"
        case temp if temp > target:
            return "cool"
        case _:
            return "hold"

print(adjust_thermostat(68, 72))

# =========================================================
# //* Codice scritto da ChatGpt (versione migliorata)
# =========================================================

def adjust_thermostat(temp, target):
    if temp < target:
        return "heat"
    elif temp > target:
        return "cool"
    else:
        return "hold"

# =========================================================
# //* Pro e contro del mio codice
# =========================================================

# //* Pro
# - Uso corretto di match-case
# - Struttura leggibile
# - Logica corretta

# //! Contro
# - Overengineering per un problema semplice
# - Stringhe non coerenti con output richiesto
# - match-case non necessario per due condizioni
# - codice meno immediato rispetto a if/elif/else

# =========================================================
# //* Consigli
# =========================================================

# - Usa if/elif/else per condizioni semplici
# - Usa match-case solo per casi complessi
# - Mantieni coerenza con i valori richiesti dall’esercizio
# - Preferisci soluzioni più semplici e leggibili quando possibile